docs/adapters/configuration.md

```diff
@@ -0,0 +1,105 @@
+## Goal
+Most simply and consistently across adapters, allow users to configure one or multiple connections in addition to a single path to the adapter module. 
+
+## Options and Problems
+
+### 1. Single connection object.
+- Eval: Simple, can include `path`, but does not allow for several connections/environments (e.g. prod/staging/dev).
+- Example: 
+```
+{
+  host: 'local',
+  port: 5432,
+  path: 'optional'
+}
+```
+
+### 2. An array of connection objects
+- Description: Each member of array has a unique `id` field so the user can indicate which connection to use.
+- Eval: Simple but we cannot assign `path` to the array unless we enforce some hacky rule to always insert it into the first element.
+- Example: 
+```
+[
+{
+  id: "default",
+  host: 'local',
+  port: 5432
+},
+{
+  id: "prod",
+  host: 'prodHost',
+  port: 5432
+}
+]
+```
+
+### 3. An object with a key `instances`
+- Description: An object with a key `instances` that holds option number 2 above but also has optional key `path`. 
+- Eval: An extra layer of nesting in a config file is less simple than option 1 and 2 and user usually won't need to specify path anyway so most of the time this structure won't be necessary.
+- Example:
+```
+{ 
+instances:[
+  {
+    id: "default",
+    host: 'local',
+    port: 5432
+  },
+  {
+    id: "prod",
+    host: 'local',
+    port: 5432
+  },
+],
+path: "path"
+}
+```
+
+### 4. An object with with unique key `id`s
+- Description: An object with a key `path` and connection objects as values behind unique `id` keys.
+- Eval: Less simple due to nesting and can be ambiguous if you allow a single connection object as well.
```

Comment:
What's the ambiguous thing? Afaict this is the approach used by https://github.com/juttle/juttle-elastic-adapter which has worked fine for me as a user so far. I didn't have to specify `path` because I don't know what that is, and the need never came up. The nesting did not strike me as difficult - I mean, our config is a JSON file, so nesting is expected... I personally find it simpler if it's just objects, not arrays of objects (one type of braces to align).